api/handlers/dataexplorerhandler.py

```diff
@@ -305,7 +305,39 @@ def _parse_request(self, request_type='search'):
         for f in filters:
             if f.get('terms'):
                 for k,v in f['terms'].iteritems():
-                    modified_filters.append({'terms': {k+'.raw': v}})
+                    if "null" in v:
+                        try:
+                            v.remove("null")
+                        except AttributeError:
+                            v = None
+                        null_filter = {
+                            'bool': {
+                                'should': [
```

Comment:
Since this is the only bool keyword at this level, it works out having this be a `should`, but `should`s are used mostly to affect the scoring and it is not required that they match (when with more strict bool options like `must`). I would switch it to a `must` so if new logic is added here, it has the intention.